doc/manual/types-vs-contracts.md

```diff
@@ -0,0 +1,148 @@
+# Type vs contracts: when to?
```

Comment:
I'm questioning whether this should be a separate document.

I think that it's difficult to introduce all the type-related stuff in the other document without explaining why they matter. Which you do a little, but is always best put in terms of “why/when do you want to use this”.

Intuitively I'd merge the two document with a structure like: here are Nickel's things (contract application, type annotation, etc…) told briefly, the whys of each, then go into the weed of what is supported and how they articulate.

My feeling is that it would flow much more nicely, and be easier to understand.

Reply:
One single document for contracts, typing and how/when to use them would be a bit too big, I think. We could split by "overview" speaking about both typing and contracts, and then have separate detailed sections about typing and contracts, though.

It's possible to inline this one into the respective documents on contracts and typing, or into the potential first overview. The rationale of having this as a separate guide is to have a kind of quick access, "cheat sheet"-like document, indexing sections by the situations, while the e.g. typing section is organized differently. I think it's not a problem if we're repeating things from the other sections.